client/src/containers/Authentication.jsx

```diff
@@ -15,28 +18,42 @@ const Authentication = () => {
   const renderForm = () => {
     if (activeItem === 'signin') {
       return <SigninForm />;
-    } else if (activeItem === 'signup') {
+    } if (activeItem === 'signup') {
       return <SignupForm />;
     }
   };
-
   return (
     <div>
-      <Menu pointing secondary style={menuStyle}>
-        <Menu.Item
-          name="Sign in"
-          active={activeItem === 'signin'}
-          onClick={() => setActiveItem('signin')}
-        />
-        <Menu.Item
-          name="Sign up"
-          active={activeItem === 'signup'}
-          onClick={() => setActiveItem('signup')}
-        />
-      </Menu>
-      {renderForm()}
+      <Modal
+        show={showModal}
+        modalClosed={toggleAuthModal}
+      >
+        <Menu pointing secondary style={menuStyle}>
+          <Menu.Item
+            name="Sign in"
+            active={activeItem === 'signin'}
+            onClick={() => setActiveItem('signin')}
+          />
+          <Menu.Item
+            name="Sign up"
+            active={activeItem === 'signup'}
+            onClick={() => setActiveItem('signup')}
+          />
+        </Menu>
+        {renderForm()}
+      </Modal>
     </div>
   );
 };
 
-export default Authentication;
+const mapStateToProps = (state) => ({
+  showModal: state.auth.showAuthModal,
+});
+const mapDispatchToProps = (dispatch) => ({
```

Comment:
you can use dictonary syntax if you have the name of the key and function are same
```suggestion
const mapDispatchToProps = {
   toggleAuthModal
};
```